Core/src/org/sleuthkit/autopsy/sqlitereader/SQLiteReader.java

```diff
@@ -0,0 +1,191 @@
+        return DriverManager.getConnection(
+                "jdbc:sqlite:" + dbPath); //NON-NLS
+    }
+    
+     /**
+     * Gets the table names and schemas from the SQLite database file.
+     *
+     * @return A mapping of table names to SQL CREATE TABLE statements.
+     */
+    public static Map<String, String> getTableNameAndSchemaPairs(Connection connection)
+            throws SQLException {
+        
+        Map<String, String> dbTablesMap = new TreeMap<>();
+        
+        try (Statement statement = connection.createStatement();
+                ResultSet resultSet = statement.executeQuery(
+                    "SELECT name, sql FROM sqlite_master " //NON-NLS
+                    + " WHERE type= 'table' " //NON-NLS
+                    + " ORDER BY name;")){ //NON-NLS
+            
+            while (resultSet.next()) {
+                String tableName = resultSet.getString("name"); //NON-NLS
+                String tableSQL = resultSet.getString("sql"); //NON-NLS
+                dbTablesMap.put(tableName, tableSQL);
+            }
+        }
+        
+        return dbTablesMap;
+    }
+    
+    public static Integer getTableRowCount(Connection connection, 
+            String tableName) throws SQLException {
+        try (Statement statement = connection.createStatement();
+                ResultSet resultSet = statement.executeQuery(
+                        "SELECT count (*) as count FROM " + tableName)){
+            return resultSet.getInt("count");
+        }
+    }
+    
+    public static List<Map<String, Object>> getRowsFromTable(Connection connection, 
+            String tableName) throws SQLException {
+        try(Statement statement = connection.createStatement();
+                ResultSet resultSet = statement.executeQuery(
+                        "SELECT * FROM " + tableName)) {
+            return resultSetToList(resultSet);
+        }
+    }
+    
+    @NbBundle.Messages({"# {0} - tableName",
+        "SQLiteExtractor.readTable.errorText=Error getting rows for table: {0}"})
+    public static List<Map<String, Object>> getRowsFromTable(Connection connection, String tableName, 
+            int startRow, int numRowsToRead) throws SQLException{
+        
+        try(Statement statement = connection.createStatement();
+                ResultSet resultSet = statement.executeQuery(
+                        "SELECT * FROM " + tableName
+                        + " LIMIT " + Integer.toString(numRowsToRead)
+                        + " OFFSET " + Integer.toString(startRow - 1))) {
+            return resultSetToList(resultSet);
+        }
+    }
+    
+    @NbBundle.Messages({"SQLiteExtractor.exportTableToCsv.write.errText=Failed to export table content to csv file.",
+                        "SQLiteExtractor.exportTableToCsv.FileName=File name: ",
+                        "SQLiteExtractor.exportTableToCsv.TableName=Table name: "
+    })
+    public static void exportTableToCSV(File file, String tableName, 
```

Comment:
I think this method should stay in the file viewer, it is specific to its use case.